servicetalk-http-api/src/main/java/io/servicetalk/http/api/Http2Settings.java

```diff
@@ -0,0 +1,84 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+
+/**
+ * Object representing a <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a> frame.
+ */
+public interface Http2Settings {
+    /**
+     * Get the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_HEADER_TABLE_SIZE</a>.
+     * @return the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_HEADER_TABLE_SIZE</a>.
+     */
+    @Nullable
+    Integer headerTableSize();
+
+    /**
+     * Get the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_CONCURRENT_STREAMS</a>.
+     * @return the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_CONCURRENT_STREAMS</a>.
+     */
+    @Nullable
+    Integer maxConcurrentStreams();
+
+    /**
+     * Get the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_INITIAL_WINDOW_SIZE</a>.
+     * @return the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_INITIAL_WINDOW_SIZE</a>.
+     */
+    @Nullable
+    Integer initialWindowSize();
+
+    /**
+     * Get the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_FRAME_SIZE</a>.
+     * @return the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_FRAME_SIZE</a>.
+     */
+    @Nullable
+    Integer maxFrameSize();
+
+    /**
+     * Get the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a>.
+     * @return the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a>.
+     */
+    @Nullable
+    Integer maxHeaderListSize();
+
+    /**
+     * Get the setting value associated with an
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">identifier</a>.
+     * @param identifier the <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">identifier</a>.
+     * @return {@code null} if no setting value corresponding {@code identifier} exists, otherwise the value.
+     */
+    @Nullable
+    Integer settingValue(char identifier);
```

Comment:
Do we currently have a use-case for a custom setting? If not, let's keep API minimal by exposing only settings from the spec that we know the behavior for. Arbitrary settings may lead to unexpected behavior or override known settings.

Reply:
I think we should leave this as there maybe arbitrary setting identifiers (not only for us to process) and we won't be able to easily add it later once the API stabilizes.

Reply:
We can add `default` method later if necessary and support by our impl. It's hard to imagine a valid use-case for a custom setting, and it opens more doors for misuse. We are not netty that have to be flexible for any purpose, the controlled environment that doesn't let users break themselves is more user-friendly.

Reply:
discussed offline ... I think it is useful to keep for now as we would ideally use the same object in the future to expose to users (e.g. for visibility in transport observer, `StreamingHttpConnection.transportEventStream`, ..). The risk it will confuse folks is minimal as for creation they will mostly go through the builder.

Reply:
Agreed, would be nice to reuse the same API for observing settings. I like that you haven't added the builder method for a custom setting yet. Only advanced users will be able to configure that by implementing `Http2Settings` contract.